backend/schedulerd/executor.go

```diff
@@ -21,41 +22,42 @@ var (
 
 // Executor executes scheduled or adhoc checks
 type Executor interface {
-	processCheck(ctx context.Context, check *types.CheckConfig) error
+	processCheck(ctx context.Context, check *corev2.CheckConfig) error
 	getEntities(ctx context.Context) ([]cache.Value, error)
-	publishProxyCheckRequests(entities []*types.Entity, check *types.CheckConfig) error
-	execute(check *types.CheckConfig) error
-	buildRequest(check *types.CheckConfig) (*types.CheckRequest, error)
+	publishProxyCheckRequests(entities []*corev2.Entity, check *corev2.CheckConfig) error
+	execute(check *corev2.CheckConfig) error
+	buildRequest(check *corev2.CheckConfig) (*corev2.CheckRequest, error)
 }
 
 // CheckExecutor executes scheduled checks in the check scheduler
 type CheckExecutor struct {
-	bus         messaging.MessageBus
-	store       store.Store
-	namespace   string
-	entityCache *cache.Resource
+	bus                    messaging.MessageBus
+	store                  store.Store
+	namespace              string
+	entityCache            *cache.Resource
+	secretsProviderManager *secrets.ProviderManager
 }
 
 // NewCheckExecutor creates a new check executor
-func NewCheckExecutor(bus messaging.MessageBus, namespace string, store store.Store, cache *cache.Resource) *CheckExecutor {
+func NewCheckExecutor(bus messaging.MessageBus, namespace string, store store.Store, cache *cache.Resource, secretsProviderManager *secrets.ProviderManager) *CheckExecutor {
 	return &CheckExecutor{bus: bus, namespace: namespace, store: store, entityCache: cache}
```

Comment:
You added `secretsProviderManager` in the parameters but never set it on the `CheckExecutor` struct, which explains why I still see a nil panic!